modules/database-migrations/lambda.tf

```diff
@@ -0,0 +1,93 @@
+resource "aws_iam_role" "lambda_assume_role" {
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role_document.json
+  name               = "TDRDbMigrationLambdaRole${title(var.environment)}"
+}
+
+data "aws_iam_policy_document" "lambda_assume_role_document" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "lambda_migration_document" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:ListBucket"
+    ]
+    resources = [
+      "arn:aws:s3:::tdr-database-migrations/*",
+      "arn:aws:s3:::tdr-database-migrations"
+    ]
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "ec2:DeleteNetworkInterface",
```

Comment:
Might be a silly question, but why does the lambda need these permissions? Is it because it's running in a private subnet?

Reply:
Yeah, if it's inside a VPC, it needs these.